### PR TITLE
#7543 Add keepdim support to moreh_sum operation

### PR DESCRIPTION
**Pull request by dongjin-na**
This PR adds keepdim support to moreh_sum and moreh_sum_backward operations.

### CODE REVIEW

File: tt_eager/tt_dnn/op_library/moreh_sum/moreh_sum_op.cpp [hunk 1]
```diff
@@ -81,69 +148,60 @@ void MorehSum::validate_with_output_tensors(
...
+        // e.g. (2, 64, 64) with dim 1 to be (2, 1[32], 64)
+        // e.g. (2, 64, 64) with dim 0 to be (64, 64)
+        for (int i = 0; i < input_rank; ++i) {
+            bool is_reduced_dim = (i == this->dim);
+            if (is_reduced_dim && !is_tile_dim)
+                continue;
+
+            shape.push_back((is_reduced_dim && is_tile_dim) ? (TILE_HEIGHT) : (input_shape[i]));
+            pad_dimensions.push_back(
+                (is_reduced_dim && is_tile_dim) ? (Padding::PadDimension{0, 31}) : (input_padding[i]));
```
Comment (namhyeong-kim) [inline tt_eager/tt_dnn/op_library/moreh_sum/moreh_sum_op.cpp:198]: What about making error when ```keep_dim=True && is_reduced_dim```?
Automatically turning off ```keep_dim``` could be confusing to users.

File: tt_eager/tt_dnn/op_library/moreh_sum/moreh_sum_op.cpp [hunk 2]
```diff
@@ -81,69 +148,60 @@ void MorehSum::validate_with_output_tensors(
...
     const auto& input_shape = input.get_legacy_shape();
     const auto input_rank = input_shape.rank();
+    const bool is_tile_dim = (this->dim == input_rank - 1 || this->dim == input_rank - 2);
+    log_debug(LogOp, "{}:{} dim {}, keepdim {}", __func__, __LINE__, this->dim, this->keepdim);
 
-    // keepdim=true
-    auto output_shape = input_shape;
-    auto padding = output_shape.padding();
+    Shape output_shape = input_shape;
+    if (this->keepdim) {
```
[the remark this answers is not in the document]
Comment (dongjin-na) [reply tt_eager/tt_dnn/op_library/moreh_sum/moreh_sum_op.cpp:169]: In that case, a shape in batch dim can't be set to 1.

**Comment by sangwon-chae** [comment]
@davorchap Would you please review this PR?

**Comment by sangwon-chae** [comment]
@tt-aho @TT-BrianLiu @mywoodstock Would you please review the PR? Thanks.

**Review by tt-aho — APPROVED** [review]

**Comment by dongjin-na** [comment]
https://github.com/tenstorrent/tt-metal/actions/runs/9509866802 passed.

**Pull request merged**